internal/storage/image.go

```diff
@@ -398,6 +401,43 @@ func (svc *imageService) ImageStatus(systemContext *types.SystemContext, nameOrI
 	return &result, nil
 }
 
+func (svc *imageService) IsRunningImageAllowed(ctx context.Context, systemContext *types.SystemContext, nameOrID string) (bool, error) {
+	ref, err := svc.getRef(nameOrID)
+	if err != nil {
+		return false, fmt.Errorf("get ref for name or ID: %w", err)
+	}
+
+	policy, err := signature.DefaultPolicy(systemContext)
+	if err != nil {
+		return false, fmt.Errorf("get default policy: %w", err)
+	}
+
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return false, fmt.Errorf("create policy context: %w", err)
+	}
+
```

Comment:
(I imagine at least the file reading+parsing in `DefaultPolicy`, and _possibly_ `NewPolicyContext`, could be pre-created or preserved across innovations. (`PolicyContext` only allows a single thread at at a time, though.))

Reply:
I added a cache around the policy context.